packages/flutter_tools/lib/src/run_hot.dart

```diff
@@ -563,6 +570,21 @@ class HotRunner extends ResidentRunner {
     );
   }
 
+  bool _isPaused() {
+    for (FlutterDevice device in flutterDevices) {
+      for (FlutterView view in device.views) {
+        if (view.uiIsolate != null) {
+          final ServiceEvent pauseEvent = view.uiIsolate.pauseEvent;
+          if (pauseEvent != null && pauseEvent.isPauseEvent) {
```

Comment:
Wouldn't `isPauseEvent` always be true for `view.uiIsolate.pauseEvent`?  Should we use an assert here instead of a condition on the `if`?